tests/src/Integration/Condition/ListContainsTest.php

```diff
@@ -139,4 +139,42 @@ public function testConditionEvaluation() {
       ->setContextValue('item', $entity_four);
     $this->assertFalse($this->condition->evaluate());
   }
+
+  /**
+   * Test refining the context definitions.
+   *
+   * @covers ::refineContextDefinitions
+   */
+  public function testRefiningContextDefinitions() {
+    // Create array of mock entities.
+    $entity_zero = $this->getMock('Drupal\Core\Entity\EntityInterface');
+    $entity_zero->expects($this->any())
+      ->method('id')
+      ->willReturn('entity_zero_id');
+    $entity_zero->expects($this->any())
+      ->method('getEntityTypeId')
+      ->willReturn('entity_zero_type_id');
+
+    $entity_one = $this->getMock('Drupal\Core\Entity\EntityInterface');
+    $entity_one->expects($this->any())
+      ->method('id')
+      ->willReturn('entity_one_id');
+    $entity_one->expects($this->any())
+      ->method('getEntityTypeId')
+      ->willReturn('entity_one_type_id');
+
+    // Test array of entities.
+    $entity_list = [$entity_zero, $entity_one];
+
+    $this->condition
+      ->setContextValue('list', $entity_list)
+      ->setContextValue('item', $entity_zero);
+    $this->condition->refineContextdefinitions();
+
+    // Get the context definition for the item
+    // and make sure that the type is now the entity type.
+    $item = $this->condition->getContextDefinition('item');
```

Comment:
item is a bad variable name here, since this is not the item but the context definition of item.